app/models/category.py

```diff
@@ -0,0 +1,103 @@
+from sqlalchemy import String, Boolean, Integer, Text, Date, DECIMAL, ForeignKey, Enum
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from uuid import UUID
+from .base import Base
+from .evaluation import ComprehensiveEvaluation
+from .user import User
+from enum import Enum as PyEnum
+from .evaluation import RecommendedChallenge
+
+
+# 카테고리 Enum 클래스 정의
+class CategoryNameEnum(PyEnum):
+    CODING = "코딩"
+    ENGLISH = "영어"
+    WORKOUT = "운동"
+
+
+# 색상 Enum 클래스 정의
+class CategoryColorEnum(PyEnum):
+    CODING = "#FDA63A"  # 호박색
+    ENGLISH = "#6C88C4"  # 인디고 블루
+    WORKOUT = "#556B2F"  # 올리브 그린
+
+
+# CategoryProgress 테이블 (한 카테고리에 여러 목표, 노트가 연결됨)
+class CategoryProgress(Base):
+    __tablename__ = "category_progresses"
+
+    user_id: Mapped[UUID] = mapped_column(
+        UUID(as_uuid=True), ForeignKey("users.id"), nullable=False
+    )
+    category_name: Mapped[CategoryNameEnum] = mapped_column(
+        Enum(CategoryNameEnum, name="category_name_enums"), nullable=False
+    )
+
+    total_goal: Mapped[int] = mapped_column(Integer, nullable=False)
+    completed_goal: Mapped[int] = mapped_column(Integer, nullable=False)
+    progress_rate: Mapped[DECIMAL] = mapped_column(DECIMAL(5, 2), default=0.00)
+
+    # CategoryProgress → User, Goal (N : 1 관계)
+    users: Mapped["User"] = relationship(
+        "User", back_populates="category_progresses", cascade="all, delete"
+    )
+    goals: Mapped[list["Goal"]] = relationship(
+        "Goal", back_populates="category_progresses", cascade="all, delete"
+    )
```

Comment:
관계가 이렇게 되는것 같은데 확인 부탁드립니당 

```suggestion
    goals: Mapped["Goal"] = relationship(
        "Goal", back_populates="category_progresses", cascade="all, delete"
    )
```

Reply:
이부분은 제가 약간 헷갈리는데 리스트를 달아주면 하나의 카테고리가 여러개의 목표를 지정할수 있게되는데,
하나의 카테고리가 하나의 목표를 바라보는게 적절한거 같긴해서 일단 수정은 해놓고 질문 남겨봅니다
@trivialcoding6 @chaeengg

Reply:
@Ryan-OH 
목표와 진행률 테이블을 말씀 하신거로 보이는데요! 기덕님이 제안하신게 리스트가 어니라 1:1 관계가 되도록 제안주신거 같은데 그럼 동준님이 말해주신거랑 일치하는거 아닌가요?!

Reply:
결국 차이는 categoryprogress가 여러개의 goal을 가질수 있냐 없나의 관점에서 list를 다느냐 안다느냐의 차이인것 같은데 확인부탁드립니다 우선 수정은 list를 제외하였습니다

Reply:
@Ryan-OH 
제가 알기로는 goals -> 영어 코딩 운동 이거로 알고있는데 각 목표별로 진행률이 여러개일수있나요..!? 혹시 제가 잘못 파악하고 있는거면 정정 부탁드립니다🙏🙏

cc @GideokKim @chaeengg

Reply:
저는 CategoryProgress → Goal가 N : 1 관계를 가진다고 ERD에서 봤는데 list가 없는게 N : 1이고 list가 있으면 1 : N 관계로 묶인다고 코드상 이해했어요.

만약 list가 없는 표현이 1:1 관계를 가진다고 하면 지금 코드 전체에 걸쳐 user table과의 관계가 N : 1이 아닌 1:1이 된다는 의미이므로 ERD랑 다르게 코드를 작성한게 되는데 어떤게 맞는건가요?

정리하자면 아래 코드가 N대1을 의미하는지 1대1을 의미하는지 알려주세요!

```python
 users: Mapped["User"] = relationship(
        "User", back_populates="category_progresses", cascade="all, delete"
    )
```

Reply:
기덕님이 이해하신계 맞아요 아래코드는 N:1관계고   list가 없는게 N : 1이고 list가 있으면 1 : N 관계로 묶인다가 맞는것 같습니다

Reply:
@trivialcoding6 ERD 상에는 CategoryProgress → Goal가 N : 1의 관계를 가진다고 ERD에 작성되어 있는거로 이해해서 코드 리뷰를 그렇게 했고 실제로 그게 가능한 건지는 생각하지 않았습니당 저는 코드와 ERD가 일치하는지만 체크하면서 코드 리뷰를 했어요! 말씀하신 부분 생각하니 저도 헷깔리는데 이 부분은 따로 정리가 필요할 것 같고 지금은 ERD랑 일치시키는 것만 목표로 코드 리뷰를 하겠습니당

@Ryan-OH 넵 그렇다면 지금 코드가 ERD와 일치한다고 보여서 이렇게 작성하는게 맞다고 생각됩니당

Reply:
@GideokKim 
아 넵넵!! 저도 생각해보니 채은님한테 동일하게 여쭤봐서 답변을 받았던거 같은데 기억이 잘 안나네요 🥲 ERD 상으론 맞는거 같아서 저도 이의 없습니다!

cc @Ryan-OH @chaeengg

Reply:
확인했고 기존에 수정했던대로 (기덕님이 제시하신 코드로) 냅두겠습니닷!